compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -330,7 +330,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         // Don't show generic arguments when the method can't be found in any implementation (#81576).
         let mut ty_str_reported = if trait_missing_method {
-            ty_str.strip_prefix("dyn ").expect("Failed to remove the prefix dyn").to_owned()
+            ty_str
+                .split_ascii_whitespace()
+                .nth(1)
+                .expect("Failed to remove prefix `dyn`")
+                .to_owned()
```

Comment:
Could you avoid string inspection, and match on `recv_ty.kind()` to extract the trait path?